core/unit_test/TestRangePolicyCTAD.cpp

```diff
@@ -17,76 +17,134 @@
 #include <Kokkos_Core.hpp>
 #include "Kokkos_Core_fwd.hpp"
 
-#if !defined(KOKKOS_COMPILER_NVCC) || KOKKOS_COMPILER_NVCC >= 1120
-
 namespace {
 
-template <class... Args>
-using PolicyMaker = decltype(::Kokkos::RangePolicy(std::declval<Args>()...));
-
-template <class Policy, class... Args>
-inline constexpr bool IsSamePolicy =
-    std::is_same_v<Policy, PolicyMaker<Args...>>;
+struct TestRangePolicyCTAD {
+  struct SomeExecutionSpace {
+    using execution_space = SomeExecutionSpace;
+    using size_type       = size_t;
 
-#define KOKKOS_TEST_RANGE_POLICY(...) static_assert(IsSamePolicy<__VA_ARGS__>)
```

Comment:
Can you try if this approach also works to make the test a little more concise?

Reply:
The current version appears to be working under all the compilers we care about.  I've checked all the logs in the CI, and all those builds are passing w/o warnings in this area.

We've been down a similar path already.  That is a material change (no instance variables) that may not work across the board.  I'm fine with trying that in the future, but not part of this P/R.

Reply:
I'm fine with a follow-up pull request for that.